Mathlib/Data/Matrix/Basic.lean

```diff
@@ -762,6 +762,25 @@ theorem sum_elim_dotProduct_sum_elim : Sum.elim u x ⬝ᵥ Sum.elim v y = u ⬝
   simp [dotProduct]
 #align matrix.sum_elim_dot_product_sum_elim Matrix.sum_elim_dotProduct_sum_elim
 
+/-- Permuting a vector on the left of a dot product can be transferred to the right. -/
+@[simp]
+theorem comp_equiv_symm_dotProduct (e : m ≃ n) : u ∘ e.symm ⬝ᵥ x = u ⬝ᵥ x ∘ e :=
+  (e.sum_comp _).symm.trans <|
+    Finset.sum_congr rfl fun _ _ => by simp only [Function.comp, Equiv.symm_apply_apply]
+#align matrix.comp_equiv_symm_dot_product Matrix.comp_equiv_symm_dotProduct
+
+/-- Permuting a vector on the right of a dot product can be transferred to the left. -/
+@[simp]
+theorem dotProduct_comp_equiv_symm (e : n ≃ m) : u ⬝ᵥ x ∘ e.symm = u ∘ e ⬝ᵥ x := by
+  simpa only [Equiv.symm_symm] using (comp_equiv_symm_dotProduct u x e.symm).symm
+#align matrix.dot_product_comp_equiv_symm Matrix.dotProduct_comp_equiv_symm
+
+/-- Permuting vectors on both sides of a dot product is a no-op. -/
+@[simp]
+theorem comp_equiv_dotProduct_comp_equiv (e : m ≃ n) : x ∘ e ⬝ᵥ y ∘ e = x ⬝ᵥ y := by
+  rw [← dotProduct_comp_equiv_symm]; simp only [Function.comp, Equiv.apply_symm_apply]
```

Comment:
This proof isn't the same as the mathport one; what went wrong? Can you add a porting note?

Reply:
With `simp only [← dotProduct_comp_equiv_symm, Function.comp, Equiv.apply_symm_apply]` I get
```
unsolved goals
l: Type ?u.146570
m: Type u_1
n: Type u_2
o: Type ?u.146579
m': o → Type ?u.146584
n': o → Type ?u.146589
R: Type ?u.146592
S: Type ?u.146595
α: Type v
β: Type w
γ: Type ?u.146602
inst✝²: Fintype m
inst✝¹: Fintype n
inst✝: NonUnitalNonAssocSemiring α
uvw: m → α
xy: n → α
e: m ≃ n
⊢ ((fun x_1 => x (↑e x_1)) ⬝ᵥ fun x => y (↑e x)) = x ⬝ᵥ y
```

Reply:
However I have added the porting note in this file anyway.

Reply:
I think the problem is that `Function.comp` is no longer reducible.

In the long run, that means we need copies of these lemmas written with and without `comp`.

I think the porting note is fine for now though.